lib/spack/spack/solver/concretize.lp

```diff
@@ -463,8 +470,11 @@ variant_default_value(Package, Variant, Value) :- variant_default_value_from_cli
 
 % Treat 'none' in a special way - it cannot be combined with other
 % values even if the variant is multi-valued
-:- 2 {variant_value(Package, Variant, Value): variant_possible_value(Package, Variant, Value)},
+:- 2 {
+     variant_value(Package, Variant, Value) : variant_possible_value(Package, Variant, Value)
+   },
    variant_value(Package, Variant, "none"),
+   build(Package),
```

Comment:
Is there a use case for which we need to restrict this to built packages? I'm not convinced we should allow externals/concrete specs to violate this.

Reply:
I could see the case for externals, but not for concrete specs in general. If we want the concrete specs to be at all reusable, we can't hold them to constraints that apply to current variants -- they may have been concretized before the package looked the way it does.  So here, you might have an old spec that has a variant value of `none,foo`.  Without this, using it for anything is unsatisfiable.  See the most recent bugfix commit as well, as that deals with a similar situation for variants and complers.

Reply:
We have never allowed `none,foo` to be a variant value, from when MV variants were first introduced. Unless I'm missing something